### PR TITLE
More CoreLib code cleanup

### PR DESCRIPTION
**Pull request by stephentoub**
In between-meetings cleanup guided by various analyzers (and mostly fixed via auto-fixers).

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Array.cs [hunk 1]
```diff
@@ -936,9 +936,8 @@ public static int IndexOf(Array array, object? value, int startIndex, int count)
                 return retVal;
 #endif
 
-            object[]? objArray = array as object[];
             int endIndex = startIndex + count;
-            if (objArray != null)
+            if (array is object[]objArray)
```
Comment (Wraith2) [inline src/System.Private.CoreLib/shared/System/Array.cs]: space needed on both changes in this file.

File: src/System.Private.CoreLib/shared/System/AggregateException.cs [hunk 2]
```diff
@@ -257,8 +257,7 @@ private AggregateException(string message, IList<ExceptionDispatchInfo> innerExc
                 throw new ArgumentNullException(nameof(info));
             }
 
-            Exception[]? innerExceptions = info.GetValue("InnerExceptions", typeof(Exception[])) as Exception[];
-            if (innerExceptions == null)
+            if (!(info.GetValue("InnerExceptions", typeof(Exception[])) is Exception[] innerExceptions))
```
Comment (AaronRobinsonMSFT) [inline src/System.Private.CoreLib/shared/System/AggregateException.cs]: To be honest, I don't see how this is more readable.

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeParse.cs [hunk 3]
```diff
@@ -5246,6 +5246,7 @@ private static void Trace(string s)
             // Internal.Console.WriteLine(s);
         }
 
+        // for testing; do not make this readonly
```
Comment (AaronRobinsonMSFT) [inline src/System.Private.CoreLib/shared/System/Globalization/DateTimeParse.cs]: Is there a suppression that can be applied for future analyzer runs?

File: src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs [hunk 4]
```diff
@@ -6659,10 +6660,9 @@ private bool TrySetFromTask(Task task, bool lookForOce)
                 case TaskStatus.Faulted:
                     var edis = task.GetExceptionDispatchInfos();
                     ExceptionDispatchInfo oceEdi;
-                    OperationCanceledException? oce;
                     if (lookForOce && edis.Count > 0 &&
                         (oceEdi = edis[0]) != null &&
-                        (oce = oceEdi.SourceException as OperationCanceledException) != null)
+                        oceEdi.SourceException is OperationCanceledException oce)
```
Comment (AaronRobinsonMSFT) [inline src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs]: I don't see `oce` being used anywhere.

File: src/System.Private.CoreLib/shared/System/Threading/Timer.cs [hunk 5]
```diff
@@ -124,7 +124,7 @@ private bool EnsureTimerFiresBy(uint requestedDuration)
         private const int ShortTimersThresholdMilliseconds = 333;
 
         // Time when Pause was called
-        private volatile int _pauseTicks = 0;
+        private readonly int _pauseTicks = 0;
```
Comment (AaronRobinsonMSFT) [inline src/System.Private.CoreLib/shared/System/Threading/Timer.cs]: Shouldn't this be `const` instead?

File: src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs (hunk 4 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs]: It's used on the next line: `oce.CancellationToken`

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeParse.cs (hunk 3 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Globalization/DateTimeParse.cs]: We could add a pragma warning disable for IDE0044, but there are other analyzers that flag for this as well.  If this is inadvertently changed, then either a) tests should fail (at least based on the comment) or b) it won't have mattered :)

File: src/System.Private.CoreLib/shared/System/AggregateException.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/AggregateException.cs]: Reverted.

File: src/System.Private.CoreLib/shared/System/Array.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Array.cs]: Fixed

File: src/System.Private.CoreLib/shared/System/Threading/Timer.cs (hunk 5 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Threading/Timer.cs]: I'm actually going to follow this up with a separate change that deletes _pauseTicks entirely.

**Pull request merged**